crates/forge_domain/src/context.rs

```diff
@@ -19,11 +19,12 @@ pub enum ContextMessage {
 }
 
 impl ContextMessage {
-    pub fn user(content: impl ToString) -> Self {
+    pub fn user(content: impl ToString, model: Option<ModelId>) -> Self {
         ContentMessage {
             role: Role::User,
             content: content.to_string(),
             tool_calls: None,
+            model,
```

Comment:
add a method on `Context::model(&self) -> Option<Model>`. Which is always going to be the last set model in the context.